record-dot-preprocessor.cabal

```diff
@@ -69,12 +69,14 @@ test-suite record-dot-preprocessor-test
         base == 4.*,
         extra,
         record-hasfield,
-        filepath
+        filepath,
+        beam-core
```

Comment:
I'd rather avoid the beam dependency if we can. Are there no types in `base` that let us express similar problems and will fail in similar ways?

Reply:
This PR is the result of a problem we had at @juspay internally, so right now references beam because it was the primary usecase — but if the approach seems good to you, the PR should definitely be polished and beam can be scrubbed away from it.

Reply:
Same re/ plugin vs preprocessor. Agreed that both should be implemented.

Reply:
Yep, I finisher preprocessor patching. But I have some problems with beam dependency removing. I have to add new library module with my own type family for testing. But as a result I have strange errors:

```
record-dot-preprocessor/utils/Utils.hs:1:1: error:
    File name does not match module name:
    Saw: ‘Main’
    Expected: ‘Utils’
```

If someone have any idea how to add new module to this project structure, could you advice me

Reply:
That sounds like you haven't written `module Utils where` at the top of the module. I'd still encourage you to look for something pre-installed with base - `Const` and `WrappedMonad` might work: https://hackage.haskell.org/package/base-4.14.0.0/docs/Control-Applicative.html#t:Const

Reply:
Well, I tried with/without module name, but it doesn't work for now. I'll try more and write more specific question later :) 

As about pre-installed base types - I don't think it is possible (probably I wrong) because the current version of the plugin should work well with constructions like
```
data MyData f = MyData {field :: f String}

instance HasField "field" (MyData f) (f String) where ...
```
Because there are no type families in the instance type

So we need some type family instead of `C` from Beam, and it should be in a separate module for tests with qualified import. But, probably we can remove tests for qualified imports of C

Reply:
I'd be tempted to just skip the qualified import tests - probably more hassle than they are worth. I think you do need the module header, but once you have that, I can imagine there are import path issues too.

Reply:
@ndmitchell Well, I tried to make it as a lib few more times. It works well when I add a utility module with C type family to the plugin folder, because it is marked as library in cabal-file, I don't think it is correct approach. But I have problems when I tried to add it into tet folder, because of using `system_ "runhaskell"` on it. So, I decided to inlne this test-type-family into Both.hs. Please take a lok at result. I think I finished all review fixes